### PR TITLE
Handle delete cluster with failed stack

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Description
Part of #2172 

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Added labels for change area (e.g. `area/nodegroup`), target version (e.g. `version/0.12.0`) and kind (e.g. `kind/improvement`)
- [x] Make sure the title of the PR is a good description that can go into the release notes

### CODE REVIEW

File: pkg/eks/eks.go [hunk 1]
```diff
@@ -195,7 +200,6 @@ func (c *ClusterProvider) CanOperate(spec *api.ClusterConfig) (bool, error) {
 	case awseks.ClusterStatusCreating, awseks.ClusterStatusDeleting, awseks.ClusterStatusFailed:
 		return false, fmt.Errorf("cannot perform Kubernetes API operations on cluster %q in %q region due to status %q", spec.Metadata.Name, spec.Metadata.Region, status)
 	default:
-		// all other states are considered operable, including UPDGRADING (which is missing from the SDK)
```
Comment (cPu1) [inline pkg/eks/eks.go:198]: The part about the `UPGRADING` state missing in the SDK was useful in this comment.

Comment (michaelbeaumont) [reply pkg/eks/eks.go:198]: Is it still missing? I see an Updating state

Comment (michaelbeaumont) [reply pkg/eks/eks.go:198]: https://github.com/aws/aws-sdk-go/blob/d85f343adb9b56f552867c2fa558d60f43a02242/service/eks/api.go#L6653-L6654

Comment (cPu1) [reply pkg/eks/eks.go:198]: It was about the `UPGRADING` state. I think the original comment might've actually meant the `UPDATING` state, as when a cluster is being upgraded, the status is `UPDATING`. So I think we're good to remove this comment.

**Pull request merged**